dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -362,25 +363,25 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
                     if (response.isPresent()) {
                         Map<String, ArrayList<Map<String, String>>> map = new HashMap<>();
                         Map<String, String> errorMap = new HashMap<>();
-                        map = (Map<String, ArrayList<Map<String, String>>>)gson.fromJson(response.get(), map.getClass());
-                        errorMap = (Map<String, String>)gson.fromJson(response.get(), errorMap.getClass());
-                        if (errorMap.get("error_message") != null) {
-                            LOG.error("Error response from Quay: " + errorMap.get("error_message"));
-                        }
-
-                        try {
-                            final List<Map<String, String>> array = map.get("tags");
-
-                            for (Map<String, String> tag : array) {
-                                final String digest = tag.get("manifest_digest");
-                                final String imageID = tag.get("image_id");
-                                List<Checksum> checksums = new ArrayList<>();
-                                checksums.add(new Checksum(digest.split(":")[0], digest.split(":")[1]));
-                                dockerImages.add(new Image(checksums, repo, tagName, imageID));
+                        map = (Map<String, ArrayList<Map<String, String>>>)GSON.fromJson(response.get(), map.getClass());
+                        errorMap = (Map<String, String>)GSON.fromJson(response.get(), errorMap.getClass());
+                        if (errorMap.get(errorKey) != null) {
+                            LOG.error("Error response from Quay: " + errorMap.get(errorKey));
+                        } else {
+                            try {
+                                final List<Map<String, String>> array = map.get("tags");
+
+                                for (Map<String, String> tag : array) {
+                                    final String digest = tag.get("manifest_digest");
+                                    final String imageID = tag.get("image_id");
+                                    List<Checksum> checksums = new ArrayList<>();
+                                    checksums.add(new Checksum(digest.split(":")[0], digest.split(":")[1]));
+                                    dockerImages.add(new Image(checksums, repo, tagName, imageID));
+                                }
+
+                            } catch (IndexOutOfBoundsException | NullPointerException ex) {
+                                LOG.error("Could not get checksum information for " + splitDocker[1]);
```

Comment:
Also, what can cause the IndexOutOfBoundsException? If it's on `splitDocker`, which on a quick skim of the code, I don't think it is, then you'll get an exception logging the exception.

Also, rather than using indexes, it would be nice to have a variable name, but not crucial.

Reply:
I don't think it'd ever really happen if the call to quay was successful, but it could happen when splitting the manifest_digest to get the checksum on line 378. But now that you mention it, someone could put an incomplete quay URL and it could fail on 357.